### PR TITLE
FilteredArrayManager refactored to prevent duplicated query

### PR DESCRIPTION
**Pull request by piotr-travisci**

### CODE REVIEW

File: app/utils/filtered-array-manager.js [hunk 1]
```diff
@@ -183,8 +183,11 @@ let FilteredArrayManager = EmberObject.extend({
     const filterArray = this.filteredArrayManagerForType(modelName).getFilterArray(queryParams, filterFunction, dependencies);
 
     if (queryParams) {
-      const queryPromise = this.store.query(modelName, queryParams);
-      const promise = resolve(queryPromise).then(() => filterArray);
+      let currentRecords = this.store.peekAll(modelName);
+      if (filterFunction) {
+        currentRecords = currentRecords.filter(record => filterFunction(record));
+      }
+      const promise = resolve(currentRecords).then(() => filterArray);
```
Comment (artursmirnov) [inline app/utils/filtered-array-manager.js]: I'm not really sure about this change. I didn't try it live, but logically it does completely different things. My concerns are:
1) It doesn't use `queryParams` even though it only runs when they exist. And previous implementation did use them
2) It never runs requests to API, always picking records from the store, not sure if it's expected. I guess, at some point it should fetch data from API, shouldn't it?
3) `filterFunction` might already be applied by `filteredArrayManagerForType`, not sure if it needs to be applied manually

Also, as a side note, I think it's safe to shorten `filterFunction` call to just `currentRecords.filter(filterFunction)`

Comment (artursmirnov) [reply app/utils/filtered-array-manager.js]: It has been discussed privately, all concerns are resolved. Looks good to me 👍

Comment (piotr-travisci) [reply app/utils/filtered-array-manager.js]: For wider visibility of implemented solution:
`const filterArray = this.filteredArrayManagerForType(modelName).getFilterArray(queryParams, filterFunction, dependencies);` code already make API call via `store.query`.
In the meantime while we are waiting for data from backend we try to show what we already have in store
`let currentRecords = this.store.peekAll(modelName);`

Comment (piotr-travisci) [reply app/utils/filtered-array-manager.js]: and many thanks for your review @artursmirnov !

**Pull request merged**